src/PkgAuthentication.jl

```diff
@@ -198,22 +199,35 @@ function step(state::NeedRefresh)::Union{HasNewToken, NoAuthentication, Failure}
     # errors are recoverable by just getting a new token:
     if response isa Downloads.Response && response.status == 200
         try
-            body = JSON.parse(String(take!(output)))
-            if haskey(body, "token")
-                return HasNewToken(state.server, body["token"])
+            body = Pkg.TOML.parse(String(take!(output)))
+            let msg = "token refresh response"
+                assert_dict_keys(body, "access_token", "id_token"; msg=msg)
+                assert_dict_keys(body, "expires_in"; msg=msg)
+                assert_dict_keys(body, "expires", "expires_at"; msg=msg)
             end
+            return HasNewToken(state.server, body)
         catch err
             @debug "invalid body received while refreshing token" exception=(err, catch_backtrace())
         end
         return NoAuthentication(state.server)
     else
-        @debug "request for refreshing token failed" exception=(err, catch_backtrace())
+        @debug "request for refreshing token failed" response
```

Comment:
This is outside of `try ... catch`, so no exception to show. Instead, I guess we can show the `response` value.